cluster-autoscaler/cloudprovider/gce/gce_manager.go

```diff
@@ -303,8 +303,15 @@ func (m *gceManagerImpl) CreateInstances(mig Mig, delta int64) error {
 	for _, ins := range instances {
 		instancesNames = append(instancesNames, ins.Id)
 	}
+	baseName, found := m.cache.GetMigBasename(mig.GceRef())
+	if !found {
+		baseName, err = m.GceService.FetchMigBasename(mig.GceRef())
+		if err != nil {
+			return fmt.Errorf("can't upscale %s: failed to collect BaseInstanceName: %w", mig.GceRef(), err)
+		}
```

Comment:
Maybe consider m.cache.SetMigBasename() here? I don't think it will change all that much, but I see no obvious reason not to update the cache if we're fetching the value anyway.